### ISSUE TITLE
Error: Debug Failure. False expression. at Object.first during 'getApplicableRefactors'

### ISSUE THREAD

**Issue by aozgaa** [Microsoft/TypeScript#21500]
**TypeScript Version Prefix**:
2.7.0
Note that this exception comes from an insiders builds. Since there are multiple insiders builds with the same version prefix, line numbers may be difficult to track down.

**VSCode Version**
1.20.0-insider
**command requested**
getApplicableRefactors
**hitting sessions**
6288
**proportion of all sessions**
0.111013029200946
**stack**
```
Error: Debug Failure. False expression.
at Object.first in tsserver.js (line: 2153, column: 15)
at collectEnclosingScopes in tsserver.js (line: 81230, column: 65)
at getPossibleExtractionsWorker in tsserver.js (line: 81312, column: 30)
at getPossibleExtractions in tsserver.js (line: 81271, column: 26)
at Object.getAvailableActions in tsserver.js (line: 80873, column: 35)
at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:77543:12 in  (line: undefined, column: )
at getIterator in tsserver.js (line: 1729, column: 23)
at Object.next in tsserver.js (line: 1724, column: 35)
at Object.arrayFrom in tsserver.js (line: 2263, column: 32)
at Object.getApplicableRefactors in tsserver.js (line: 77542, column: 23)
at Object.getApplicableRefactors in tsserver.js (line: 83782, column: 32)
at IOSession.Session.getApplicableRefactors in tsserver.js (line: 90159, column: 53)
at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) in tsserver.js (line: 89128, column: 61)
at IOSession.Session.executeWithRequestId in tsserver.js (line: 90349, column: 28)
at IOSession.Session.executeCommand in tsserver.js (line: 90358, column: 33)
at IOSession.Session.onMessage in tsserver.js (line: 90378, column: 35)
at Interface.<anonymous> in tsserver.js (line: 91579, column: 27)
at emitOne in events.js (line: 96, column: 13)
at Interface.emit in events.js (line: 191, column: 7)
at Interface._onLine in readline.js (line: 241, column: 10)
at Interface._normalWrite in readline.js (line: 384, column: 12)
at Socket.ondata in readline.js (line: 101, column: 10)
at emitOne in events.js (line: 96, column: 13)
at Socket.emit in events.js (line: 191, column: 7)
at readableAddChunk in _stream_readable.js (line: 178, column: 18)
at Socket.Readable.push in _stream_readable.js (line: 136, column: 10)
at Pipe.onread in net.js (line: 560, column: 20)
```
EDIT: updated

**Comment by amcasey** [comment]
Woohoo!  This one's not about line numbers. :smile:

**Comment by amcasey** [comment]
@aozgaa, is this affecting 7% of sessions or 0.07% of sessions?

**Comment by amcasey** [comment]
@aozgaa, do we sanitize stacks and, if so, could this indicate a bug in our sanitizer?

`at tsserver.js:77539:12 in  (line: undefined, column: )`

**Comment by aozgaa** [comment]
7%, though I noticed there was additional de-deuplication to be done. The actual number, updated immediately after posting this, is 11%.

**Comment by amcasey** [comment]
That's crazy.  This is 2.7.0rc?  Or an internal build?  I'm having some trouble lining up the stack with the source.

**Comment by aozgaa** [comment]
That's a bug in the sanitizer I wrote to post-process the query results.

**Comment by amcasey** [comment]
Oh, this appears to be a dup of https://github.com/Microsoft/TypeScript/issues/20559.  The fix is in `release-2.7`, but not in the RC.
